### PR TITLE
Add pytorch3d

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
Note that I disabled tests on CI as a physical GPU is required.

I ran the tests locally with a GPU - those are the results:
```
Ran 867 tests in 144.388s

FAILED (failures=2, errors=12, skipped=45)
```

And here is a summary of the failures/errors, none of them are to worry about:
```
FileNotFoundError: [Errno 2] No such file or directory: '/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/docs/tutorials/data/cow_mesh/cow.obj'

  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/implicitron/test_extending_orm_types.py", line 90, in build
    frame_data = super().build(frame_annotation, sequence_annotation, load_blobs)
TypeError: build() takes 3 positional arguments but 4 were given

ValueError: Can't write images with one color channel.

ImportError: PyCUDA was compiled without GL extension support
(which then leads to:)
ImportError: cannot import name 'MeshRasterizerOpenGL' from 'pytorch3d.renderer.opengl'

AssertionError: "Faces must have at least 3 vertices." does not match "Not enough face data."
```

<details><summary>Detailed test failure log</summary>
```
======================================================================
ERROR: test_simple (tests.implicitron.test_data_cow.TestDataCow)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/implicitron/test_data_cow.py", line 28, in test_simple
    self._runtest(use_point_light=True, num_views=4)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/implicitron/test_data_cow.py", line 32, in _runtest
    provider = RenderedMeshDatasetMapProvider(**kwargs)
  File "<string>", line 12, in __init__
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/implicitron/dataset/rendered_mesh_dataset_map_provider.py", line 111, in __post_init__
    mesh = io.load_mesh(data_file, device=device)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/io/pluggable.py", line 128, in load_mesh
    mesh = mesh_interpreter.read(
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/io/obj_io.py", line 314, in read
    mesh = load_objs_as_meshes(
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/io/obj_io.py", line 265, in load_objs_as_meshes
    verts, faces, aux = load_obj(
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/io/obj_io.py", line 223, in load_obj
    with _open_file(f, path_manager, "r") as f:
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/io/utils.py", line 25, in _open_file
    f = path_manager.open(f, mode)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/iopath/common/file_io.py", line 1023, in open
    bret = handler._open(path, mode, buffering=buffering, **kwargs)  # type: ignore
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/iopath/common/file_io.py", line 611, in _open
    return open(  # type: ignore
FileNotFoundError: [Errno 2] No such file or directory: '/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/docs/tutorials/data/cow_mesh/cow.obj'

======================================================================
ERROR: test_extending_frame_data (tests.implicitron.test_extending_orm_types.TestExtendingOrmTypes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/implicitron/test_extending_orm_types.py", line 203, in test_extending_frame_data
    item = dataset[i]
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/implicitron/dataset/sql_dataset.py", line 179, in __getitem__
    return self._get_item(frame_idx, True)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/implicitron/dataset/sql_dataset.py", line 235, in _get_item
    frame_data = self.frame_data_builder.build(
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/implicitron/test_extending_orm_types.py", line 90, in build
    frame_data = super().build(frame_annotation, sequence_annotation, load_blobs)
TypeError: build() takes 3 positional arguments but 4 were given

======================================================================
ERROR: test_basic_1chan (tests.pulsar.test_forward.TestForward)
Test rendering one image with one sphere, 1 channel.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/pulsar/test_forward.py", line 191, in test_basic_1chan
    imageio.imsave(
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/imageio/v2.py", line 397, in imwrite
    return file.write(im, **kwargs)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/imageio/plugins/pillow.py", line 405, in write
    raise ValueError("Can't write images with one color channel.")
ValueError: Can't write images with one color channel.

======================================================================
ERROR: test_basic_3chan (tests.pulsar.test_forward.TestForward)
Test rendering one image with one sphere, 3 channels.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/pulsar/test_forward.py", line 136, in test_basic_3chan
    imageio.imsave(
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/imageio/v2.py", line 397, in imwrite
    return file.write(im, **kwargs)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/imageio/plugins/pillow.py", line 405, in write
    raise ValueError("Can't write images with one color channel.")
ValueError: Can't write images with one color channel.

======================================================================
ERROR: test_basic_8chan (tests.pulsar.test_forward.TestForward)
Test rendering one image with one sphere, 8 channels.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/pulsar/test_forward.py", line 261, in test_basic_8chan
    imageio.imsave(
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/imageio/v2.py", line 397, in imwrite
    return file.write(im, **kwargs)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/imageio/plugins/pillow.py", line 405, in write
    raise ValueError("Can't write images with one color channel.")
ValueError: Can't write images with one color channel.

======================================================================
ERROR: test_bg_weight (tests.pulsar.test_forward.TestForward)
Test background reweighting.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/pulsar/test_forward.py", line 73, in test_bg_weight
    imageio.imsave(
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/imageio/v2.py", line 397, in imwrite
    return file.write(im, **kwargs)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/imageio/plugins/pillow.py", line 405, in write
    raise ValueError("Can't write images with one color channel.")
ValueError: Can't write images with one color channel.

======================================================================
ERROR: test_principal_point (tests.pulsar.test_forward.TestForward)
Test shifting the principal point.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/pulsar/test_forward.py", line 313, in test_principal_point
    imageio.imsave(
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/imageio/v2.py", line 397, in imwrite
    return file.write(im, **kwargs)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/imageio/plugins/pillow.py", line 405, in write
    raise ValueError("Can't write images with one color channel.")
ValueError: Can't write images with one color channel.

======================================================================
ERROR: test_load_lumpy (tests.test_io_off.TestMeshOffIO)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/test_io_off.py", line 137, in test_load_lumpy
    mesh = io.load_mesh(f.name)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/io/pluggable.py", line 128, in load_mesh
    mesh = mesh_interpreter.read(
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/io/off_io.py", line 436, in read
    data = _load_off_stream(f)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/io/off_io.py", line 260, in _load_off_stream
    faces, faces_colors = _read_faces(file, n_faces)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/io/off_io.py", line 143, in _read_faces
    faces_ncolors_colors = _read_faces_lump(file, n_faces=n_faces, n_colors=None)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/io/off_io.py", line 88, in _read_faces_lump
    raise ValueError("Not enough face data.") from None
ValueError: Not enough face data.

======================================================================
ERROR: tests.test_rasterize_rectangle_images (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.test_rasterize_rectangle_images
Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/test_rasterize_rectangle_images.py", line 32, in <module>
    from pytorch3d.renderer.opengl import MeshRasterizerOpenGL
ImportError: cannot import name 'MeshRasterizerOpenGL' from 'pytorch3d.renderer.opengl' (/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/renderer/opengl/__init__.py)


======================================================================
ERROR: tests.test_rasterizer (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.test_rasterizer
Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/test_rasterizer.py", line 25, in <module>
    from pytorch3d.renderer.opengl.rasterizer_opengl import (
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/renderer/opengl/rasterizer_opengl.py", line 16, in <module>
    import pycuda.gl
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pycuda/gl/__init__.py", line 5, in <module>
    raise ImportError("PyCUDA was compiled without GL extension support")
ImportError: PyCUDA was compiled without GL extension support


======================================================================
ERROR: tests.test_render_meshes (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.test_render_meshes
Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/test_render_meshes.py", line 49, in <module>
    from pytorch3d.renderer.opengl import MeshRasterizerOpenGL
ImportError: cannot import name 'MeshRasterizerOpenGL' from 'pytorch3d.renderer.opengl' (/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/renderer/opengl/__init__.py)


======================================================================
ERROR: tests.test_render_multigpu (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.test_render_multigpu
Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/test_render_multigpu.py", line 28, in <module>
    from pytorch3d.renderer.opengl import MeshRasterizerOpenGL
ImportError: cannot import name 'MeshRasterizerOpenGL' from 'pytorch3d.renderer.opengl' (/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.9/site-packages/pytorch3d/renderer/opengl/__init__.py)


======================================================================
FAIL: test_bad (tests.test_io_off.TestMeshOffIO)
----------------------------------------------------------------------
ValueError: Not enough face data.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/test_io_off.py", line 278, in test_bad
    load(lines2)
AssertionError: "Faces must have at least 3 vertices." does not match "Not enough face data."

======================================================================
FAIL: test_no_egl_error (tests.test_opengl_utils.TestOpenGLUtils)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tobias/mambaforge/conda-bld/pytorch3d_1688623721552/test_tmp/tests/test_opengl_utils.py", line 380, in test_no_egl_error
    self.assertTrue(_can_import_egl_and_pycuda())
AssertionError: False is not true

----------------------------------------------------------------------
Ran 867 tests in 144.388s

FAILED (failures=2, errors=12, skipped=45)
In thread 0, device 0.
In thread 1, device 1.
In thread 2, device 0.
In thread 3, device 1.
In thread 4, device 0.
In thread 5, device 1.
In thread 6, device 0.
In thread 7, device 1.
In thread 8, device 0.
In thread 9, device 1.
EGL could not release context on device cuda:0. This can happen if you created two contexts on the same device. Instead, you can use DeviceContextStore to use a single context per device, and EGLContext.make_(in)active_in_current_thread to (in)activate the context as needed.
EGL could not release context on device cuda:0. This can happen if you created two contexts on the same device. Instead, you can use DeviceContextStore to use a single context per device, and EGLContext.make_(in)active_in_current_thread to (in)activate the context as needed.
```
</details>

**Comment by Tobias-Fischer** [comment]
Hi @conda-forge/help-python-c this is ready for review. Many things are borrowed from the torchvision and other feedstocks. Thanks a lot for reviewing in advance!

For context, a lot of my people here in the QUT Centre for Robotics have been struggling installing this package - I imagine it’s the same elsewhere. Hopefully this will make their lives easier :)

/cc @conda-forge/pytorch-cpu 
/cc @wolfv - this is heavily used in the robotic vision community

**Comment by Tobias-Fischer** [comment]
Thanks for the feedback @ngam! I've added the `pip check` - makes sense having it :)

**Comment by Tobias-Fischer** [comment]
And pytorch3d sits on top of pytorch - similar to torchvision, torchkin etc. sitting on top of pytorch.

File: recipes/pytorch3d/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    - '*/libtorch_cpu{{ SHLIB_EXT }}'
+    - '*/libtorch_cuda{{ SHLIB_EXT }}'
+    - '*/libtorch_python{{ SHLIB_EXT }}'
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}                 # [cuda_compiler_version != "None"]
+    - sysroot_linux-64 ==2.17                # [linux64]
```
Comment (wolfv) [inline recipes/pytorch3d/meta.yaml:44]: should this be `==` (because I think that only matches 2.17.0.0...`

Comment (hmaarrfk) [reply recipes/pytorch3d/meta.yaml:44]: Correct the equal signs should be removed according to the docs.

https://conda-forge.org/docs/maintainer/knowledge_base.html#using-centos-7

Comment (Tobias-Fischer) [reply recipes/pytorch3d/meta.yaml]: Thanks for the catch!

File: recipes/pytorch3d/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    - pytorch                                # [build_platform != target_platform]
+    - pytorch =*={{ torch_proc_type }}*      # [build_platform != target_platform]
+  host:
+    - python
+    - pip
+    - setuptools
+    - cudnn                                  # [cuda_compiler_version != "None"]
+    - numpy
+    - pytorch
+    - pytorch =*={{ torch_proc_type }}*
```
Comment (wolfv) [inline recipes/pytorch3d/meta.yaml:57]: I also think the initial `=*` could be shortened to `*`, but prob. doesn't matter much.

Comment (Tobias-Fischer) [reply recipes/pytorch3d/meta.yaml:57]: I’ve seen it this way in quite a few repos - would suggest to keep as it is battle tested :)

**Comment by Tobias-Fischer** [comment]
Hi @wolfv - would you like to see any more changes?

File: recipes/pytorch3d/meta.yaml (hunk 2 above)
Comment (Tobias-Fischer) [reply recipes/pytorch3d/meta.yaml:57]: Ref: https://github.com/search?q=org%3Aconda-forge%20pytorch%20%3D*%3D%7B%7B%20torch_proc_type%20%7D%7D*&type=code

In other repos, it's always used as `=*=`, never as `*=`

**Comment by Tobias-Fischer** [comment]
Kind reminder @wolfv @conda-forge/help-python-c :)

**Pull request merged**